### ISSUE TITLE
Remove warning for missing setting from local.setting.json file in case they are present in user secrets file.

### ISSUE THREAD

**Issue by soninaren** [Azure/azure-functions-core-tools#2219]
Some information in this PR https://github.com/Azure/azure-functions-core-tools/pull/2136

Repro posted by @vijayrkn 
> Function:
> 
> ```
>     public static class Function
>     {
>         [FunctionName("Function")]
>         public static void Run([QueueTrigger("myqueue-items", Connection = "ConnectionStrings:MyQueueConn")]string myQueueItem, ILogger log)
>         {
>             log.LogInformation($"C# Queue trigger function processed: {myQueueItem}");
>         }
>     }
> ```
> 
> local.settings.json
> 
> ```
> {
>     "IsEncrypted": false,
>     "Values": {
>         "AzureWebJobsStorage": "UseDevelopmentStorage=true",
>         "FUNCTIONS_WORKER_RUNTIME": "dotnet"
>     }
> }
> ```
> 
> user secrets:
> 
> ```
> {
>   "ConnectionStrings:MyQueueConn": "UseDevelopmentStorage=true"
> }
> ```
> 
> csproj:
> 
> ```
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <TargetFramework>netcoreapp3.1</TargetFramework>
>     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
>     <UserSecretsId>4a9f61f7-0f22-4d2d-bc40-6ef82a58df03</UserSecretsId>
>   </PropertyGroup>
>   <ItemGroup>
>     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="3.0.10" />
>     <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="3.1.0" />
>     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.7" />
>   </ItemGroup>
>   <ItemGroup>
>     <None Update="host.json">
>       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
>     </None>
>     <None Update="local.settings.json">
>       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
>       <CopyToPublishDirectory>Never</CopyToPublishDirectory>
>     </None>
>   </ItemGroup>
> </Project>
> ```
> 
> Let me know if you need anything more.

**Comment by soninaren** [comment]
@gzuber This is the follow up workitem from this issue: https://github.com/Azure/azure-functions-core-tools/issues/1970

Also please cherry pick the change here to dev branch: https://github.com/Azure/azure-functions-core-tools/pull/2136

**Comment by gzuber** [comment]
After some testing, this is also a problem if user specifies "AzureWebJobsStorage" in user-secrets. It will throw an exception and not attempt to start the function.

**Comment by SeanFeldman** [comment]
The User Secret file is useless for both the triggers and host required information (storage account connection string) expected to be in the `local.settings.json` file. This is blocking me from using User Secrets properly.

**Comment by gzuber** [comment]
Hey @SeanFeldman -- I just did some testing and was able to set up a few different functions where the trigger connection string was in my user secrets file. There's a misleading warning that makes it seem like it won't work, but it does (which is what this issue is intended to address). I think I would need some more info about your app and what you're trying to do to help you out.

Like I commented above, specifying "AzureWebJobsStorage" in user secrets is broken unfortunately because it throws an exception but I'll be fixing that with this work item.
